### ISSUE TITLE
Tuple member is incorrectly inferred as Unknown

### ISSUE THREAD

**Issue by michelbl** [microsoft/pyright#6119]
**Describe the bug**

The following function seems to me to be well typed:

```python
def foo(n: int, s0: str, s2: str):
    list_of_tuples: list[tuple[str, str, str]] = []
    for _ in range(0):
        if n == 0:
            print(list_of_tuples[-1][0])
            last = list_of_tuples[-1]
            list_of_tuples[-1] = (s0, last[1], s2)
```

However, pyright 1.1.331 reports the following error:
```
<filepath>:6:13 - error: Type of "last" is partially unknown
    Type of "last" is "tuple[str, str, str] | tuple[str, str | Unknown, str]" (reportUnknownVariableType)`
```

Swapping lines 5 and 6 solves the error, though:

```python
def bar(n: int, s0: str, s2: str):
    list_of_tuples: list[tuple[str, str, str]] = []
    for _ in range(0):
        if n == 0:
            last = list_of_tuples[-1]
            print(list_of_tuples[-1][0])
            list_of_tuples[-1] = (s0, last[1], s2)
```

The example seems to be complex, but if I remove the `for` instruction, or if I remove the `if` instruction, or if I remove one element from the tuple, it suppresses the error.

**VS Code extension or command-line**

The problem appear using pyright CLI 1.1.331 and pyright through pylance (version 1.1.328)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by michelbl** [comment]
Wow, that fix was fast! :bullettrain_front: 
Thanks!

**Comment by erictraut** [comment]
This is included in pyright 1.1.332, which I just published. It will also be included in a future release of pylance.
